LoopKit/InsulinKit/HKQuantitySample+InsulinKit.swift

```diff
@@ -92,6 +122,24 @@ extension HKQuantitySample {
         return metadata?[MetadataKeyProgrammedTempBasalRate] as? HKQuantity
     }
 
+    var insulinModel: InsulinModel? {
+        guard let rawType = metadata?[MetadataKeyInsulinCurveType] as? Int, let modelType = InsulinModelType(rawValue: rawType) else {
+            return nil
+        }
+        
+        var model: InsulinModel? = nil
+        switch modelType {
+        case .walsh:
+            model = WalshInsulinModel(actionDuration: metadata?[MetadataKeyInsulinCurveDuration] as! TimeInterval, delay: (metadata?[MetadataKeyInsulinCurveDelay] ?? 600) as! TimeInterval)
+        case .exponential:
+            model = ExponentialInsulinModel(actionDuration: metadata?[MetadataKeyInsulinCurveDuration] as! TimeInterval, peakActivityTime: metadata?[MetadataKeyInsulinCurvePeak] as! TimeInterval, delay: (metadata?[MetadataKeyInsulinCurveDelay] ?? 600.0) as! TimeInterval)
+        default:
+            break
+        }
+        
+        return model
```

Comment:
I'm guessing there's a shorter way to do this that I'm not thinking about - is it possible to return from the switch statement directly?